crates/forge_main/src/commit/handler.rs

```diff
@@ -0,0 +1,68 @@
+use std::sync::Arc;
+
+use anyhow::{Context, Result};
+use forge_api::API;
+
+use crate::cli::CommitCommandGroup;
+
+/// Handler for commit command operations
+pub struct CommitHandler<A> {
+    api: Arc<A>,
+}
+
+impl<A> CommitHandler<A> {
+    /// Creates a new CommitHandler
+    pub fn new(api: Arc<A>) -> Self {
+        Self { api }
+    }
+}
+
+impl<A: API> CommitHandler<A> {
+    /// Handle commit command
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - Git commands fail
+    /// - AI generation fails
+    /// - Commit operation fails
+    pub async fn handle(&self, args: CommitCommandGroup) -> Result<String> {
+        let commit_message = self.api.generate_commit_message(args.max_diff_size).await?;
+        if args.preview {
+            // Just return the message for preview
+            Ok(commit_message)
+        } else {
+            let cwd = self.api.environment().cwd;
+            // Check if there are staged changes first
+            let status_result = self
+                .api
+                .execute_shell_command("git diff --cached --quiet", cwd.clone(), false)
```

Comment:
Lets consolidate the logic at one place. Right now the actual commit operation is happening in forge_main, but generation is hapening in forge_app. I think exposing an api to commit directly via API would be better.